### PR TITLE
[bugfix] fix deepseek memory bug

### PR DESCRIPTION
**Pull request by zzzzwwjj**

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -908,11 +909,14 @@ def _process_reqs(
...
             attn_state = AscendAttentionState.PrefillCacheHit
 
-        attn_mask = self._make_attention_mask(seq_lens=seq_lens,
-                                              query_lens=num_scheduled_tokens,
-                                              position=positions,
-                                              attn_state=attn_state)
-        self.attn_mask = attn_mask
+        # NOTE: when use ring_mla, attn_mask don't need to generate here.
+        if not self.use_ring_mla or attn_state == AscendAttentionState.PrefillNoCache:
+            attn_mask = self._make_attention_mask(
```
Comment (ApsarasX) [inline vllm_ascend/worker/model_runner_v1.py:914]: Can a compressed mask be used? If the sequence is too long, it might cause memory waste here.

Refer https://github.com/vllm-project/vllm-ascend/pull/1100

Comment (zzzzwwjj) [reply vllm_ascend/worker/model_runner_v1.py:914]: ringmla's mask size is equal to chunksize, it won't cause memory waste.

**Comment by ganyi1996ppo** [comment]
Looks good, please cherry-pick this change back to main

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -350,6 +350,7 @@ def __init__(self, vllm_config: VllmConfig, device: torch.device):
         self.torchair_graph_enabled = ascend_config.torchair_graph_config.enabled and self.vllm_config.model_config.use_mla
         self.use_cached_npu_graph = ascend_config.torchair_graph_config.use_cached_graph
         self.torchair_graph_batch_sizes = ascend_config.torchair_graph_config.graph_batch_sizes
+        self.use_ring_mla = ascend_config.chunked_prefill_for_mla
```
Comment (jianzs) [inline vllm_ascend/worker/model_runner_v1.py:353]: Should we consider which scheduler is be using?

```python
self.use_ring_mla = ascend_config.chunked_prefill_for_mla or \
            not ascend_config.ascend_scheduler_config.enabled
```
